### PR TITLE
Add RTCOfferOptions.reofferOptions.

### PR DESCRIPTION
**Pull request by pthatcherg**
An idea for how to allow control of when do reoffer codecs that have been excluded by the remote description.

***
<a href="https://pr-preview.s3.amazonaws.com/pthatcherg/webrtc-pc/pull/988.html" title="Last updated on Apr 21, 2018, 1:42 AM GMT (5799ff5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/988/35725f7...pthatcherg:5799ff5.html" title="Last updated on Apr 21, 2018, 1:42 AM GMT (5799ff5)">Diff</a>

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -450,6 +451,15 @@
                 <code><a data-link-for=
                 "RTCPeerConnection">localDescription</a></code> attribute.</p>
               </dd>
+              <dt><dfn><code>reofferCodecs</code></dfn> of type <span class=
+              "idlMemberType"><a>boolean</a></span>, defaulting to
+              <code>false</code></dt>
+              <dd>
+                <p>When the value of this dictionary member is true,
+                the generated description will include codecs that
+                were excluded by the current remote description, as
```
Comment (taylor-b) [inline webrtc.html]: Technically, the codecs are restricted by the last applied answer, not the current remote description.

File: webrtc.html [hunk 2]
```diff
@@ -450,6 +451,15 @@
...
                 "RTCPeerConnection">localDescription</a></code> attribute.</p>
               </dd>
+              <dt><dfn><code>reofferCodecs</code></dfn> of type <span class=
+              "idlMemberType"><a>boolean</a></span>, defaulting to
+              <code>false</code></dt>
+              <dd>
+                <p>When the value of this dictionary member is true,
+                the generated description will include codecs even if
+                they were excluded by the most recent answer, as
+                defined in [[!JSEP]].</p>
```
Comment (dontcallmedom) [inline webrtc.html]: this should link to a specific section of JSEP with `<span data-jsep=section-id>[[!JSEP]]</span>`.

**Comment by alvestrand** [comment]
WG decision was to not do this at this time - if JSEP comes up with some question for which this is the answer, we may reopen it.

**Comment by taylor-b** [comment]
Was going to reopen this due to discussion on #1838, but now realizing that "reofferCodecs" actually happens by default in JSEP since draft 19. What's missing is "create answer with codecs not in offer".

**Pull request closed without merging**